HTML WIdget: Added migration guide, updated screenshot (#4117)
SoftwareAG/c8y-docs: content/cockpit/widgets-collection.md

Existing widgets based on AngularJS will automatically fall back to legacy mode, which maintains backward compatibility and allows these widgets to continue functioning. This fallback displays JavaScript code that enables legacy widget execution. However, it is strongly recommended to migrate these widgets to the new Lit-based framework as soon as possible, since AngularJS support is deprecated.
{{< /c8y-admon-important >}}

> #### To migrate a legacy widget {#to-migrate-a-legacy-widget}

Reviewer: Not sure if we have some convention regarding this, but `to-migrate-a-legacy-widget` seems a bit to generic and could be placed multiple times on this page for other widgets..

```suggestion
#### To migrate a legacy widget {#html-to-migrate-a-legacy-widget}
```

Migrating a legacy widget to the new format requires familiarity with JavaScript and HTML. The migration process involves replacing all AngularJS-specific code.

Depending on the complexity of your original widget, there are two migration approaches:

* **Simple mode**: Replace placeholders and template syntax. Use this mode if your widget does not contain custom JavaScript logic in a `<script>` tag.
* **Advanced mode**: Build complex web components with custom JavaScript. Use this mode if your widget contains custom JavaScript logic or event handlers.

**Step 1: Assess your widget**

Check your current widget code. If it contains specific JavaScript logic in a `<script>` tag, you need to use advanced mode. If not, use simple mode.

**Step 2: Migrate placeholders in simple mode**

> For example, the following legacy widget code:

Reviewer 2: ```suggestion
The following example shows legacy widget code:
```

```html
<p ng-if="device.name === 'Alpine Hiker #1'">
  Hello {{device.name}}.
</p>
```

Migrated to the template syntax, this is:
```js

> ${ this.c8yContext.name === 'Alpine Hiker #1' ? html`<p>Hello ${this.c8yContext.device}</p>` : '' }

Reviewer: ```suggestion
${ this.c8yContext.name === 'Alpine Hiker #1' ? html`<p>Hello ${this.c8yContext.name}</p>` : '' }
```

  static properties = {
    // The managed object this widget is assigned to. Can be null.
    c8yContext: { type: Object },
  };

  constructor() {
    super();
  }

  render() {
    return html`
      <style>
        ${styleImports}
      </style>
      ${this.c8yContext.name === 'Alpine Hiker #1' ? html`<p>Hello ${this.c8yContext.device}</p>` : ''}
    `;
  }
}
```

This example demonstrates the basic structure of a Lit web component.

**Step 5: Add interactivity in advanced mode**

In advanced mode, you add elements like event handlers. This allows you to migrate more complex legacy widgets with custom `<script>` tags. The following example is shortened to show only the important parts:

```js

//[…]

  static properties = {
    c8yContext: { type: Object },
    helloText: { type: String } // add this to ensure "rerendering" on changes
  };

  //[…]

  sayHello() {
   this.helloText = 'Hello world';
  }

  render() {
    return html`
      <style>
        ${styleImports}
      </style>
      ${this.c8yContext.name === 'Alpine Hiker #1' ? html`<p @click="${this.sayHello}">
         Hello ${this.c8yContext.device}</p>
      ` : ''}

      ${this.helloText}
    `;
  }
//[…]
```

> This example shows how to handle click events and update component state.

Reviewer 2: ... update "the" component state? 
or
... update component states?
```suggestion
This example shows how to handle click events and update the component state.
```

**Step 6: Verify the migration**

After migration, test the widget to ensure:

* All data displays correctly.
* Interactive elements work as expected.

> * The widget responds to device or asset selection.

Reviewer 2: ```suggestion
* The widget responds to the device or asset selection.
```

> By leveraging the [lit-webcomponents](https://lit.dev/docs/v3/) framework, you migrate nearly every custom AngularJS or JavaScript implementation in a legacy widget. To request additional data, import the `fetch` library to directly request data from the platform.

Reviewer 2: ```suggestion
By leveraging the [Lit web components](https://lit.dev/docs/v3/) framework, you migrate nearly every custom AngularJS or JavaScript implementation in a legacy widget. To request additional data, import the `fetch` library to directly request data from the platform.
```
